x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table/use_initialize_rules_table_saved_state.test.ts

```diff
@@ -0,0 +1,410 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { renderHook } from '@testing-library/react-hooks';
+import { useKibana } from '../../../../../common/lib/kibana';
+import { useInitializeUrlParam } from '../../../../../common/utils/global_query_string';
+import { RULES_TABLE_MAX_PAGE_SIZE } from '../../../../../../common/constants';
+import { useInitializeRulesTableSavedState } from './use_initialize_rules_table_saved_state';
+import type { RulesTableActions } from './rules_table_context';
+import {
+  INITIAL_FILTER_OPTIONS,
+  INITIAL_SORTING_OPTIONS,
+  useRulesTableContext,
+} from './rules_table_context';
+import type { RulesTableSavedState } from './rules_table_saved_state';
+
+jest.mock('../../../../../common/lib/kibana');
+jest.mock('../../../../../common/utils/global_query_string');
+jest.mock('./rules_table_context', () => ({
+  INITIAL_FILTER_OPTIONS: jest.requireActual('./rules_table_context').INITIAL_FILTER_OPTIONS,
+  INITIAL_SORTING_OPTIONS: jest.requireActual('./rules_table_context').INITIAL_SORTING_OPTIONS,
+  useRulesTableContext: jest.fn(),
+}));
+
+function mockState({
+  urlState,
+  storageState,
+}: {
+  urlState: RulesTableSavedState | null;
+  storageState: RulesTableSavedState | null;
+}): void {
```

Comment:
Nit: arrow function

```
const mockState = ({
  urlState,
  storageState,
}: {
  urlState: RulesTableSavedState | null;
  storageState: RulesTableSavedState | null;
}): void => {
```

Reply:
It's rather a matter of taste. I prefer a `function` over an `arrow function` in the global context since it's verbose and easy to read. As a side effect `this` is `undefined` is strict mode.